src/RedemptionIdle.sol

```diff
@@ -66,6 +62,21 @@ contract RedemptionIdle is Redemption {
         });
     }
 
+    /// @notice The ```redeem``` function allows users to redeem SUPERSTATE_TOKEN for USDC at the current oracle price
+    /// @dev Will revert if oracle data is stale or there is not enough USDC in the contract
+    /// @param to The receiver address for the redeemed USDC
+    /// @param superstateTokenInAmount The amount of SUPERSTATE_TOKEN to redeem
+    function redeem(address to, uint256 superstateTokenInAmount) external override {
```

Comment:
I think you can put redeem with 3 args and redeem with 2 args fully implemented on redemption.sol instead of copy pasting into both idle and yield

Reply:
then you jsut implement _redeem in idle and yield

Reply:
https://github.com/superstateinc/onchain-redemptions/pull/37/commits/9d68bcf95ee9b59b078b3327fa0d9836651719da